libcxx/include/__algorithm/count_if.h

```diff
@@ -10,26 +10,52 @@
 #ifndef _LIBCPP___ALGORITHM_COUNT_IF_H
 #define _LIBCPP___ALGORITHM_COUNT_IF_H
 
+#include <__algorithm/for_each.h>
```

Comment:
Not attached: This optimization should also be applied to `ranges::count_if`.

Reply:
I just want to confirm that you mean to say `ranges::count_if` and not `std::count`?
Also, Is `ranges::for_each` optimizeded for segmented iterators?

Reply:
I don't know whether `ranges::for_each` is optimized. If not, we should definitely do it.